src/libpriv/rpmostree-kernel.cxx

```diff
@@ -416,18 +416,22 @@ rpmostree_finalize_kernel (int rootfs_dfd, const char *bootdir, const char *kver
       if (contents == NULL)
         return FALSE;
 
-      /* rather than trying to parse and understand the *sum format, just hackily replace */
-      g_autofree char *old_path = g_strconcat ("  /boot/vmlinuz-", kver, NULL);
-      g_autofree char *new_path = g_strconcat ("  vmlinuz-", kver, NULL);
-      g_autofree char *new_contents = rpmostree_str_replace (contents, old_path, new_path, error);
+      /* hack: fallback for when the shasum file contains a full path */
+
+      g_auto (GStrv) split_contents = g_strsplit (contents, "  ", 2);
+      if (g_strv_length (split_contents) != 2)
+        return glnx_throw (error, "Missing filename in .vmlinuz.hmac: %s", contents);
+
+      g_auto (GStrv) path_components = g_strsplit (split_contents[1], "/", -1);
+      guint path_components_length = g_strv_length (path_components);
+      if (strlen (path_components[path_components_length - 1]) == 0)
+        return glnx_throw (error, "Path in .vmlinuz.hmac is a directory: %s", contents);
+
+      g_autofree char *new_contents = g_strconcat (
+          split_contents[0], "  ", path_components[path_components_length - 1], NULL);
       if (!new_contents)
```

Comment:
This code is unnecessary now - `g_strconcat()` is an infallible operation (glib just aborts on OOM).